### PR TITLE
Replace `OicUserProperty` with `LastGrantedAuthoritiesProperty`

### PR DESCRIPTION
**Pull request by jglick**
More radical simplification of #332.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java [hunk 1]
```diff
@@ -776,60 +773,33 @@ public String getAuthenticationGatewayUrl() {
...
-                                SecurityListener.fireAuthenticated2(userDetails);
-                                return token;
-                            } else {
-                                throw new BadCredentialsException("Wrong username and password: " + authentication);
-                            }
+        return new SecurityComponents(new AuthenticationManager() {
+            public Authentication authenticate(Authentication authentication) throws AuthenticationException {
+                if (authentication instanceof AnonymousAuthenticationToken) return authentication;
+
+                if (authentication instanceof UsernamePasswordAuthenticationToken && escapeHatchEnabled) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java:780]: (ignore WS)

File: src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java [hunk 2]
```diff
@@ -776,60 +773,33 @@ public String getAuthenticationGatewayUrl() {
...
+                                new UsernamePasswordAuthenticationToken(escapeHatchUsername, "", grantedAuthorities);
+                        SecurityContextHolder.getContext().setAuthentication(token);
+                        OicUserDetails userDetails = new OicUserDetails(escapeHatchUsername, grantedAuthorities);
+                        SecurityListener.fireAuthenticated2(userDetails);
+                        return token;
+                    } else {
+                        throw new BadCredentialsException("Wrong username and password: " + authentication);
                     }
-                },
-                new UserDetailsService() {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java:808]: Switching to `SecurityComponents(AuthenticationManager)` constructor. `UserDetailsServiceProxy` throws `UserMayOrMayNotExistException2`; `ImpersonatingUserDetailsService2` handles that by checking `LastGrantedAuthoritiesProperty`.

**Review by Vlatombe — APPROVED** [review]

**Review by michael-doubez — APPROVED** [review]
LGTM

**Comment by michael-doubez** [comment]
@jglick sorry, I didn't realize #332 and #333 would conflict. Can you please resolve conflicts ?

**Comment by michael-doubez** [comment]
Many thanks

**Comment by basil** [comment]
Causes #391

**Pull request merged**